### PR TITLE
added task-in-progress and correlationId logic

### PR DESCRIPTION
**Pull request by ab48917**
Pull Request - Feature
----
 Sharding logic using correlationId as AccountId
 addTaskInProgress logic  - Anytime task is pushed to queue, the same will be added to task-in-progress table and will be there until task remains in progress. Once its done than it will be removed from queue as well as table.

**NOTE**: Please remember to run `./gradlew spotlessApply` to fix any format violations.

### CODE REVIEW

File: cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraExecutionDAO.java [hunk 1]
```diff
@@ -656,8 +694,18 @@ public long getPendingWorkflowCount(String workflowName) {
      */
     @Override
     public long getInProgressTaskCount(String taskDefName) {
-        throw new UnsupportedOperationException(
-                "This method is not implemented in CassandraExecutionDAO. Please use ExecutionDAOFacade instead.");
+        try{
+            ResultSet resultSet;
+            recordCassandraDaoRequests("getInProgressTaskCount", "n/a", taskDefName);
+            resultSet = session.execute(selectCountFromTaskInProgressStatement.bind(taskDefName));
+            return resultSet.all().size();
```
Comment (prabhakaranE6157) [inline cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraExecutionDAO.java:701]:
```suggestion
            recordCassandraDaoRequests("getInProgressTaskCount", "n/a", taskDefName);
            ResultSet resultSet = session.execute(selectCountFromTaskInProgressStatement.bind(taskDefName));
            return resultSet.all().size();
```

File: cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraExecutionDAO.java [hunk 2]
```diff
@@ -678,13 +726,35 @@ public List<WorkflowModel> getWorkflowsByType(
     @Override
     public List<WorkflowModel> getWorkflowsByCorrelationId(
             String workflowName, String correlationId, boolean includeTasks) {
-        throw new UnsupportedOperationException(
-                "This method is not implemented in CassandraExecutionDAO. Please use ExecutionDAOFacade instead.");
+        try{
+            ResultSet resultSet;
+            recordCassandraDaoRequests("getWorkflowsByCorrelationId", "n/a", correlationId);
+
+            resultSet = session.execute(selectWorkflowsByCorIdFromWorkflowStatement.bind(Integer.parseInt(correlationId)));
```
Comment (prabhakaranE6157) [inline cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraExecutionDAO.java:733]:
```suggestion
            recordCassandraDaoRequests("getWorkflowsByCorrelationId", "n/a", correlationId);

            ResultSet resultSet = session.execute(selectWorkflowsByCorIdFromWorkflowStatement.bind(Integer.parseInt(correlationId)));
```

**Review by prabhakaranE6157 — APPROVED** [review]

File: cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraExecutionDAO.java (hunk 1 above)
Comment (ab48917) [reply cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraExecutionDAO.java]: Resolved

**Review by ritikesh — APPROVED** [review]

File: cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraBaseDAO.java [hunk 3]
```diff
@@ -144,10 +146,22 @@ private String getCreateTaskLookupTableStatement() {
...
     }
 
+    /**
+     * @return cql statement to create workflow_lookup table for shard_mapping
+     */
+    private String getCreateWorkflowLookupTableStatement() {
+        return SchemaBuilder.createTable(properties.getKeyspace(), TABLE_WORKFLOW_LOOKUP)
+                .ifNotExists()
+                .addPartitionKey(WORKFLOW_ID_KEY, DataType.uuid())
+                .addColumn(SHARD_ID_KEY,DataType.cint())
```
[the remark this answers is not in the document]
Comment (ab48917) [reply cassandra-persistence/src/main/java/com/netflix/conductor/cassandra/dao/CassandraBaseDAO.java:161]: shard_id is cint (typically signed int) We are considering accountId as shard key

**Pull request merged**